Update plugins install docs for v1.11.0 (#12995)
hashicorp/packer: website/content/docs/configure.mdx

- **Windows**: If `PACKER_CONFIG_DIR` is set to `C:/`, then the configuration directory is `C:/packer.d/`. Packer does not check other values.

<a id="packer-s-config-file"></a>

##  Install a JSON configuration file

This installation method is deprecated since 1.7. Instead, use the [HCL2
`required_plugins` block](/packer/docs/templates/hcl_templates/blocks/packer#specifying-plugin-requirements) to configure how Packer installs plugins. 

You use a JSON file to set core Packer settings. Packer checks the following paths for the configuration file:

| Unix                             | Windows                           |
| -------------------------------- | --------------------------------- |
| `${PACKER_CONFIG}`               | `%PACKER_CONFIG%`                 |
| `${HOME}/.packerconfig`          | `%APPDATA%\packer.config\`        |

### JSON configuration file reference 

Use basic JSON to configure the file. You can specify the configuration parameters for the core
configuration file. All parameters are optional. None of these are required, since all have defaults.

> - `plugin_min_port`: Number that specifies the lowest port that Packer can use for communicating with plugins. Packer communicates with plugins over TCP connections on your local Unix host. Default is `10000`. We recommend setting a wide range between `plugin_min_port` and `plugin_max_port` so that Packer has access to at least 25 ports on a single run.

Author: ```suggestion
- `plugin_min_port`: Number that specifies the lowest port that Packer can use for communicating with plugins. Packer communicates with plugins over TCP or Unix sockets on your local host. Default is `10000`. We recommend setting a wide range between `plugin_min_port` and `plugin_max_port` so that Packer has access to at least 25 ports on a single run.
```
Ah, OK. Thanks for clarifying.

- `plugin_max_port`: Number that specifies highest port that Packer can for communicating with plugins. Packer communicates with plugins over TCP connections on your local host. Default is `25000`. We recommend setting a wide range between `plugin_min_port` and `plugin_max_port` so that Packer has access to at least 25 ports on a single run.
- `builders`: Specifies a type of object that installs plugins. Refer to [Builders](/packer/docs/builders) for additional information. 
- `commands`: Specifies a type of object that installs plugins. 
- `post-processors`: Specifies a type of object that installs plugins. Refer to [Post-Processors](/packer/docs/post-processors) for additional information.

> - `provisioners`: Specifies a type of object that installs plugins. Refer to [Provisioners](/packer/docs/provisioners) for additional information. 

Author: Is this comment directed at the statement about the `packer init` command in line 60? Are you suggesting that we remove it?

> - `provisioners`: Specifies a type of object that installs plugins. Refer to [Provisioners](/packer/docs/provisioners) for additional information. 

Reviewer: Not related to `packer init` per se, but `config.json` (the legacy config file) used to make it possible to declare which plugins were required by the template. That was JSON-only, and only for declaring mono-component plugins usage.

Since we don't discover them anymore on other parts of the code, and the config file is deprecated anyway, I would indeed suggest we remove references to components here. So only document the general configuration options, and leave the plugin installation/discovery to the modern alternatives.

I'm building a PR that will remove that path and explicitly error if they're encountered from the config file, so users will know these won't be used at all.
Of course this implementation may change by the time we release, depending on @nywilken's feedback then when he's back into office.

> - `provisioners`: Specifies a type of object that installs plugins. Refer to [Provisioners](/packer/docs/provisioners) for additional information. 

Reviewer: https://github.com/hashicorp/packer/pull/12998
^ The related PR here

> - `provisioners`: Specifies a type of object that installs plugins. Refer to [Provisioners](/packer/docs/provisioners) for additional information. 

Author: OK, just to be clear, we should remove the entire `## Install a JSON configuration file` section, correct? Or are you saying that we also need to remove the `## Install a configuration file` section, too? In effect, the only way to configure Packer is to export the environment variables described in `## Configuration reference`?

> - `provisioners`: Specifies a type of object that installs plugins. Refer to [Provisioners](/packer/docs/provisioners) for additional information. 

Reviewer: I imagine `commands` is a leftover from an old spec, I don't see it in the structure we deserialise into

> - `provisioners`: Specifies a type of object that installs plugins. Refer to [Provisioners](/packer/docs/provisioners) for additional information. 

Reviewer: Should we remove this section entierly then given this is out-of-date, or should we do it in a subsquent PR?

> - `builders`: Specifies a type of object that installs plugins. Refer to [Builders](/packer/docs/builders) for additional information. 
> - `commands`: Specifies a type of object that installs plugins. 
> - `post-processors`: Specifies a type of object that installs plugins. Refer to [Post-Processors](/packer/docs/post-processors) for additional information.
> - `provisioners`: Specifies a type of object that installs plugins. Refer to [Provisioners](/packer/docs/provisioners) for additional information. 

Reviewer 2: Let's remove it. 
```suggestion

```

The [`packer init`](/packer/docs/commands/init) command takes precedence over JSON-configure settings when installing plugins.

## Configure the cache directory

> Packer uses a cache directory to download large files and for logistics around large file download. By default, Packer caches files in the `./packer_cache/` folder of the current directory. Use the `PACKER_CACHE_DIR` environment variable to configure the cache directory. Refer to [`PACKER_CACHE_DIR`](/packer/docs/configure#packer_config_dir) for details.

Reviewer: ```suggestion
Packer uses a cache directory to download large files and for logistics around large file download. By default, Packer caches files in the `.packer_cache` folder of the current directory. Use the `PACKER_CACHE_DIR` environment variable to configure the cache directory. Refer to [`PACKER_CACHE_DIR`](/packer/docs/configure#packer_config_dir) for details.
```

Use basic JSON to configure the file. You can specify the configuration parameters for the core
configuration file. All parameters are optional. None of these are required, since all have defaults.

> - `plugin_min_port`: Number that specifies the lowest port that Packer can use for communicating with plugins. Packer communicates with plugins over TCP connections on your local Unix host. Default is `10000`. We recommend setting a wide range between `plugin_min_port` and `plugin_max_port` so that Packer has access to at least 25 ports on a single run.

Reviewer: For clarification, my initial comment was that the type of connection is either using TCP or UNIX sockets, depending on the OS and configuration. On Windows TCP is always used, while on UNIX systems it defaults to UNIX sockets, but you can use TCP instead (though I can't remember how you set it)
